src/plugins/config/TraceSamplerConfig.ts

```diff
@@ -44,6 +49,14 @@ class TraceSamplerConfig {
             isSampled = isSampled || this.errorAwareSampler.isSampled(span);
         }
 
+        if (this.customSampler) {
+            if (this.runCustomSamplerOnEachSpan) {
+                isSampled = true;
```

Comment:
Why we didn't call custom sampler and set `isSampled` to `true` directly even it might be set to `false` (decided not to be sampled) above by `durationAwareSampler` and/or `errorAwareSampler`?

Reply:
if `this.runCustomSamplerOnEachSpan` is true.  We run custom sampler on each span including root span. We skip running on custom sampler on root span by returning `isSampled = true;`. The running custom sampler on root span along with all spans are done in trace plugin here. https://github.com/thundra-io/thundra-lambda-agent-nodejs/pull/50/files/a44c6f9885e16034752d8d6133eeef67ccd3c6ef..187d3d3000bc7216416a2f4c290cd71a9f79a41e#diff-9a3803680d4e920a1c7139b6469e0613R201

Basically, I did it to prevent running custom sampler on root span twice.